### PR TITLE
WIP Add option to update config in dgc.sh 

### PR DESCRIPTION
**Pull request by amitaekbote**
https://mesosphere.atlassian.net/browse/DCOS-9888

### CODE REVIEW

File: dcos_installer/backend.py [hunk 1]
```diff
@@ -396,3 +396,27 @@ def success(config: Config):
     msgs['master_count'] = len(master_ips)
     msgs['agent_count'] = len(agent_ips)
     return msgs, code
+
+
+def update_config():
+    """
+    :param config_path: path to config.yaml
```
Comment (cmaloney) [inline dcos_installer/backend.py]: where are the params? :P

File: dcos_installer/cli.py [hunk 2]
```diff
@@ -155,7 +155,7 @@ def do_uninstall(*args, **kwargs):
         'Run the web interface'),
     'genconf': (
         lambda args: backend.do_configure(),
-        'EXECUTING CONFIGURATION GENERATION'
+        'EXECUTING CONFIGURATION GENERATION',
```
Comment (amitaekbote) [inline dcos_installer/cli.py]: will go away once i rebase

File: dcos_installer/backend.py [hunk 3]
```diff
@@ -393,3 +391,22 @@ def success(config: Config):
...
+    if len(validate_gen) > 0:
+        for key, error in validate_gen.items():
+            log.error('{}: {}'.format(key, error))
+        return 1
+
+    config_util.do_configure(config, updating_config=True)
+    return 0
+
+
+def generate_upgrade_script():
```
Comment (amitaekbote) [inline dcos_installer/backend.py]: Placeholder for now, will remove it once branden reviews it

File: dcos_installer/config_util.py [hunk 4]
```diff
@@ -7,21 +7,29 @@
...
 log = logging.getLogger(__name__)
 
 
-def do_configure(config):
+def do_configure(config, updating_config=False):
     gen_out = config.do_gen_configure()
-    subprocess.check_call(['mkdir', '-p', SERVE_DIR])
-    gen.installer.bash.generate(gen_out, SERVE_DIR)
+
+    config_id = gen_out.arguments['config_id']
```
Comment (amitaekbote) [inline dcos_installer/config_util.py]: All names are debatable.

File: tests/test_installer_backend.py [hunk 5]
```diff
@@ -105,7 +105,27 @@ def test_do_validate_config(tmpdir):
...
+    expected_output = {
+        'ip_detect_contents': 'ip-detect script `genconf/ip-detect` must exist',
+        'ssh_user': 'Must set ssh_user, no way to calculate value.',
+        'master_list': 'Must set master_list, no way to calculate value.',
+        'ssh_key_path': 'could not find ssh private key: genconf/ssh_key'
+    }
+    assert Config(config_path=temp_config_path).do_validate(include_ssh=True) == expected_output
+
+
+def test_update_config(tmpdir):
```
Comment (amitaekbote) [inline tests/test_installer_backend.py]: doesnt do anything of use right now

**Comment by amitaekbote** [comment]
@cmaloney @branden  good to go for first pass 

bash dcos_generate_config.sh --help 
..
--generate-upgrade-script
                        EXECUTING UPGRADE SCRIPT GENERATION
  --genconf             EXECUTING CONFIGURATION GENERATION
  --version
  --validate-config     VALIDATING CONFIGURATION
  --aws-cloudformation  EXECUTING AWS CLOUD FORMATION TEMPLATE GENERATION
  --web                 Starting DC/OS installer in web mode
  --update-config       EXECUTING CONFIGURATION GENERATION
...


bash dcos_generate_config.sh --update-config
Extracting image from this script and loading into docker daemon, this step can take a few minutes
dcos-genconf.9a85c2fac635fff1fe-e369c6d2a5450dcfce.tar
Loaded image: mesosphere/dcos-genconf:9a85c2fac635fff1fe-e369c6d2a5450dcfce
====> EXECUTING CONFIGURATION GENERATION

......

(dcos) amita@amita-desktop:~/Testing/genconf$ ls
cluster_packages.json  config.yaml  generated-config  ip-detect  serve  state 

(dcos) amita@amita-desktop:~/Testing/genconf/generated-config$ ls -l
total 8
drwxrwxrwx 3 root root 4096 Oct 26 17:39 1e08ef266878a3926a6e578e9e1199b1f135d0fd
drwxrwxrwx 3 root root 4096 Oct 26 17:38 475b5fccc070f2aaeb71863823af5f594ba5f7a9 

(dcos) amita@amita-desktop:~/Testing/genconf/generated-config/1e08ef266878a3926a6e578e9e1199b1f135d0fd$ ls
cluster_packages.json  serve

**Comment by cmaloney** [comment]
@amitaekbote please rebase on master to get the `--help` fix

**Comment by amitaekbote** [comment]
Closing in favor of https://github.com/dcos/dcos/pull/969

**Pull request closed without merging**